src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,227 @@
+
+    private static void executeTaskCase(String input, int starting_index, TaskType type) {
+        String description = input.strip().substring(starting_index);
+        try {
+            addTask(description, type);
+            printTaskMessage();
+
+        } catch (DukeException error) {
+            printErrorMessage(MISSING_ARGUMENTS_FOR_EVENT_AND_DEADLINE);
+        }
+    }
+
+    private static String[] parseInputForDifferentTask(String input,TaskType type) {
+        String[] parsedOutput = {};
+        switch(type) {
+        case TODO:
+            parsedOutput = new String[]{input};
+            break;
+        case DEADLINE:
+            parsedOutput = input.split(DEADLINE_DESCRIPTION_AND_DATE_SPLITTER);
+            break;
+        case EVENT:
+            parsedOutput = input.split(EVENT_DESCRIPTION_AND_DATE_TIME_SPLITTER);
+            break;
+        }
+        return parsedOutput;
+    }
+
+    private static void addTask(String input,TaskType type) throws StringIndexOutOfBoundsException, DukeException {
+        String[] parsedOutput = parseInputForDifferentTask(input,type);
+        switch(type){
+        case TODO:
+            tasks[Task.getTotalTasks()] = new ToDo(parsedOutput[0]);
+            break;
+        case EVENT:
+            if(parsedOutput.length < 2) {
+                throw new DukeException();
+            }
+            tasks[Task.getTotalTasks()] = new Event(parsedOutput[0], parsedOutput[1]);
+            break;
+        case DEADLINE:
+            if(parsedOutput.length < 2) {
+                throw new DukeException();
+            }
+            tasks[Task.getTotalTasks()] = new Deadline(parsedOutput[0], parsedOutput[1]);
+            break;
+        }
+        Task.setTotalTasks(Task.getTotalTasks() + 1);
+    }
+
+    private static String notifyNumberOfTasks() {
+        return "Now you have " + Task.getTotalTasks() + " task(s) in the list";
+    }
+
+    private static void executeDoneCase(String input) throws NullPointerException,DukeException,
+            ArrayIndexOutOfBoundsException, NumberFormatException{
+        if(input.length() < 6) {
+            throw new DukeException();
+        }
+        int index = Integer.parseInt(input.split(" ")[1]) - 1;
+        tasks[index].markAsDone();
+        printTaskCompletedMessage(tasks[index]);
+    }
+
+    private static void printTaskCompletedMessage(Task task) {
+        printIndentationAndDivider();
+        printWordsWithIndentation(TASK_COMPLETED_MESSAGE);
+        printWordsWithIndentation(task.getStatusIconAndDescription());
+        printIndentationAndDivider();
+        System.out.println();
+    }
+
+    private static void executeListCase() {
+        printIndentationAndDivider();
+        if(Task.getTotalTasks() == 0) {
+            printWordsWithIndentation(LIST_IS_EMPTY);
+        }
+        for (int i = 0; i < Task.getTotalTasks(); i++) {
+            printWordsWithIndentation(i + 1 + "." + tasks[i].getStatusIconAndDescription());
```

Comment:
Suggest to relocate expression in function argument to a separate line with appropriate comments.